js/NavMainComponent.js

```diff
@@ -18,161 +21,205 @@ NavMainComponent.prototype.init = function () {
         throw new Error('window must be passed to NavMainComponent');
     }
 
+    component.$body = this.$html.find('body');
     component.$navMain = this.$html.find('.nav-main');
     component.$contentMain = this.$html.find('.content-main');
     component.$navPrimary = this.$html.find('.nav-primary');
     component.$navSecondary = this.$html.find('.nav-secondary');
     component.$navTertiary = component.$navMain.find('.nav-tertiary');
     component.$navQuaternary = component.$navMain.find('.nav-quaternary');
     component.$primaryNavLinks = component.$navPrimary.find('.nav-link');
-    component.$secondaryNavLinks = component.$navSecondary.find('.nav-link');
     component.$tertiaryNavLinks = component.$navTertiary.find('.nav-link');
-    component.$quaternaryNavLinks = component.$navQuaternary.find('.nav-link');
-    component.$moreIcon = component.$navMain.find('.more-icon');
     component.$closeLink = component.$navMain.find('[data-nav-action=close]');
+    component.$mobileMenuButton = this.$html.find('.mobile-menu-button');
 
-    component.moreIconClickHandler();
+    // Calculate what primary nav items can be shown and which need to be hidden in more menu
     component.adjustNavItems();
 
-    component.$window.resize(function () {
-        component.adjustNavItems();
-    });
+    // Open navigation on mobile
+    component.$mobileMenuButton.on('click', function() {
+        component.$body.toggleClass('open-nav');
+        $(this).toggleClass('open');
 
-    component.closeHandler = function () {
-        component.closeNavs();
-    };
+        if ($(this).text() === 'Menu') {
+            $(this).text('Close');
+            $(this).attr('aria-expanded', 'true');
+        } else {
+            $(this).text('Menu');
+            $(this).attr('aria-expanded', 'false');
+        }
+    });
```

Comment:
micro optimisation here, could we cache `$(this)`, we're creating a lot of object. As far as I know `$(this) === $(this) // false`